app/code/Magento/InventoryBundleIndexer/Indexer/SourceItem/GetBundleChildrenSourceItemsIdsWithSku.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryBundleIndexer\Indexer\SourceItem;
+
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Catalog\Model\Product\Type as ProductType;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Select;
+use Magento\Inventory\Model\ResourceModel\SourceItem;
+
+/**
+ * Get bundle children source item ids by source item ids.
+ *
+ * Find product by source item sku and try find parent bundle product. Then find all children source
+ * items for this bundle and return them with bundle sku.
+ */
+class GetBundleChildrenSourceItemsIdsWithSku
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var BundleChildrenSourceItemsIdsSelectProvider
+     */
+    private $bundleChildrenSourceItemsIdsSelectProvider;
+
+    /**
+     * GetBundleChildrenSourceItemsIdsWithSku constructor.
+     *
+     * @param ResourceConnection $resourceConnection
+     * @param BundleChildrenSourceItemsIdsSelectProvider $bundleChildrenSourceItemsIdsSelectProvider
+     */
+    public function __construct(
+        ResourceConnection $resourceConnection,
+        BundleChildrenSourceItemsIdsSelectProvider $bundleChildrenSourceItemsIdsSelectProvider
+    ) {
+        $this->resourceConnection = $resourceConnection;
+        $this->bundleChildrenSourceItemsIdsSelectProvider = $bundleChildrenSourceItemsIdsSelectProvider;
+    }
+
+    /**
+     * @param array $sourceItemsIds
+     *
+     * @return array
+     */
+    public function execute(array $sourceItemsIds): array
+    {
+        $bundleIdsSelect = $this->getBundleIdsSelect($sourceItemsIds);
+
+        $select = $this->bundleChildrenSourceItemsIdsSelectProvider->execute();
+        $select->where('relation.parent_id in (?)', $bundleIdsSelect);
```

Comment:
Looks like we don't need to use subselect
We have same logic in the main select

`SELECT `source_item`.`source_item_id`, `bundle_product`.`sku`
FROM `inventory_source_item` AS `source_item`
INNER JOIN `catalog_product_entity` AS `product` ON source_item.sku = product.sku
INNER JOIN `catalog_product_relation` AS `relation` ON relation.child_id = product.entity_id
INNER JOIN `catalog_product_entity` AS `bundle_product` ON bundle_product.entity_id = relation.parent_id
WHERE (relation.parent_id IN ((

SELECT DISTINCT `bundle_product`.`entity_id`
FROM `inventory_source_item` AS `source_item`
INNER JOIN `catalog_product_entity` AS `product` ON source_item.sku = product.sku
INNER JOIN `catalog_product_relation` AS `relation` ON product.entity_id = relation.child_id
INNER JOIN `catalog_product_entity` AS `bundle_product` ON bundle_product.entity_id = relation.parent_id
WHERE (source_item.source_item_id IN (8)) AND (bundle_product.type_id = 'bundle'))))`

Reply:
We need it.